### ISSUE TITLE
Hide the top toolbar when typing

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#15189]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when typing by the fact that the button option stuff heavy top toolbar is there looking at me, like a mean supervisor.

**Describe the solution you'd like**
It would be lovely if once I start typing the top toolbar would disappear, exhibiting the same behaviour that the block toolbar has: type in block hide, stop typing show. Also the publish button could become a neutral grey when typing instead of that heavy blue.

**Describe alternatives you've considered**
 An option would be to reduce it's opacity instead of hiding. This could also just happen if I am in focus mode.

Here is how it looks with opacity .4 vs full on:

<img width="930" alt="Screenshot 2019-04-25 at 20 15 45" src="https://user-images.githubusercontent.com/107534/56755022-31847e00-6797-11e9-9ff3-e73ec381f50e.png">

<img width="925" alt="Screenshot 2019-04-25 at 20 16 49" src="https://user-images.githubusercontent.com/107534/56755054-44974e00-6797-11e9-8465-7dcf4ab71c4a.png">

@mtias do you think this would be a good addition to focus mode?

**Comment by draganescu** [comment]
As dev feedback: I explored a quick solution and it is not yet trivial because the observe typing component is not yet available when the header component is called.

**Comment by afercia** [comment]
One of the main accessibility issues in the Gutenberg UI is about controls continuously appearing/disappearing. That's a big problem for users with accessibility needs. I'd strongly recommend to not implement this in the default UI.

If a "lighter" UI is desirable, there's already the "Spotlight Mode" option in the main menu and I'd tend to think any fancy opacity / controls disappearing perfectly fit in that option :)

Note about process: please use the "Needs Accessibility Label" when an issue needs initial accessibility feedback. Instead, the "Accessibility" label should be used to track issues where initial feedback has already been given. The difference is not trivial, as we use these labels to generate reports and organize bug-scrubs.

**Comment by draganescu** [comment]
@mapk would something like this work then:

![hide-toolbar-iaw](https://user-images.githubusercontent.com/107534/61876332-4935e480-aef5-11e9-8db9-1e5d64f5ac2e.gif)

- disappear on scroll
- appear on hover or KBD shortcut

**Comment by draganescu** [comment]
I will close this PR as the introduction of the typewriter effect makes it, in my opinion, not needed.
